eZ/Bundle/EzPublishCoreBundle/Fragment/FragmentListenerFactory.php

```diff
@@ -23,12 +23,15 @@ class FragmentListenerFactory
 
     public function buildFragmentListener(UriSigner $uriSigner, $fragmentPath, $fragmentListenerClass)
     {
-        // Ensure that current pathinfo ends with configured fragment path.
-        // If so, consider it as the fragment path.
-        // This ensures to have URI siteaccess compatible fragment paths.
-        $pathInfo = $this->getCurrentRequest()->getPathInfo();
-        if (substr($pathInfo, -strlen($fragmentPath)) === $fragmentPath) {
-            $fragmentPath = $pathInfo;
+        $request = $this->getCurrentRequest();
```

Comment:
I'd rather clarify the whole thing by moving the `fragmentPath` update to a private method. What about 
```
private function updateFragmentPathIfNeeded(&$fragmentPath)
{
  if (!$request = $this->getCurrentRequest()) {
    return;
  }
  // ... do the thing
}
``` ?

It would return the